File: BusCatalog.Api/BusCatalog.Api/Domain/HealthCheck/Ports/HealthCheckResponse.cs
using System.Net;
using Microsoft.AspNetCore.Mvc;

namespace BusCatalog.Api.Domain.HealthCheck.Ports;

public record HealthCheckResponse
{
    public string? Url { get; set;}
    public DateTime Time { get; set; }
    public int StatusCode { get; set; }
    public bool Success { get; set; }
    
    public static HealthCheckResponse Generate(string url, int statusCode, bool success)
    {
        return new HealthCheckResponse
        {
            Time = DateTime.UtcNow,
            Url = url,
            StatusCode = statusCode,
            Success = success
        };
    }

    public ObjectResult ToResult()
    {
        return new ObjectResult(this)
        {
            StatusCode = Success ? StatusCode : (int)HttpStatusCode.BadGateway,
            ContentTypes = { "application/json" }
        };
    }
}

File: BusCatalog.Scraper/src/main/java/br/com/marcomvidal/buscatalog/scraper/healthcheck/ports/HealthCheckResponse.java
package br.com.marcomvidal.buscatalog.scraper.healthcheck.ports;

import java.time.LocalDateTime;

import org.springframework.http.ResponseEntity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

@AllArgsConstructor
@Getter
@Setter
public class HealthCheckResponse {
    private String url;
    private LocalDateTime time;
    private int statusCode;
    private boolean success;

    public static HealthCheckResponse generate(String url, int statusCode, boolean success) {
        return new HealthCheckResponse(
            url, 
            LocalDateTime.now(), 
            statusCode, 
            success);
    }

    public ResponseEntity<HealthCheckResponse> toResult() {
        return ResponseEntity
            .status(this.statusCode)
            .body(this);
    }
}
